vtm/src/org/oscim/tiling/source/mapfile/MapDatabase.java

```diff
@@ -717,35 +719,57 @@ private boolean processWayDataBlock(MapElement e, boolean doubleDeltaEncoding, b
             /* each way node consists of latitude and longitude */
             int len = numWayNodes * 2;
 
-            wayLengths[coordinateBlock] = decodeWayNodes(doubleDeltaEncoding,
-                    e, len, isLine);
-
+            GeoPoint[] waySegment = null;
```

Comment:
Can use a `List` here:
- For convenience and less new arrays
- Fixing also the reverse geocode case

Reply:
changed the implementation. and no need for copying the array any more.